citesphere-model/src/main/java/edu/asu/diging/citesphere/data/bib/impl/CitationMongoDao.java

```diff
@@ -72,4 +86,58 @@ public List<? extends ICitation> findCitationsInCollection(String groupId, Strin
         return mongoTemplate.find(query, Citation.class);
     }
     
+    @Override
+    public Persons findAllPeople(String groupId) {
+        MatchOperation match = Aggregation.match(Criteria.where("group").is(groupId));
+
+        ProjectionOperation project = Aggregation.project().and("group").as("_id").and("authors")
+                .unionArrays("editors", "otherCreators.person").as("persons");
+
+        UnwindOperation unwind = Aggregation.unwind("persons");
+
+        ProjectionOperation project1 = Aggregation
+                .project().and("_id").as("_id").and("persons").as("persons").and(ConditionalOperators
+                        .when(Criteria.where("$persons.uri").is(null)).then("").otherwiseValueOf("persons.uri"))
+                .as("uri");
+
+        GroupOperation group = Aggregation.group("_id").addToSet(new BasicDBObject() {
+            {
+                put("name", "$persons.name");
+                put("firstName", "$persons.firstName");
+                put("lastName", "$persons.lastName");
+                put("uri", "$uri");
+            }
+        }).as("persons");
+
+        UnwindOperation unwind1 = Aggregation.unwind("persons");
+
+        SortOperation sort = Aggregation.sort(Direction.ASC, "persons.name");
+
+        GroupOperation groupAfterSort = Aggregation.group("_id").push("persons").as("persons");
+
+        AggregationResults<Persons> result = mongoTemplate.aggregate(
+                Aggregation.newAggregation(match, project, unwind, project1, group, unwind1, sort, groupAfterSort),
+                Citation.class, Persons.class);
+
+        return result.getUniqueMappedResult();
```

Comment:
this seems very involved, can't you just use: `addField` (add all people in one array), `$unwind` (one citation per person in the person array), `$group` (group by person uri), and then sort? The first three seem to work, I haven't tried sorting though.